src/utils.rs

```diff
@@ -126,3 +127,39 @@ pub fn extract_gzip(gzip_path: &str, filename: &str, prefix: &str) -> Result<()>
     );
     Ok(())
 }
+
+/// Try and decode a base64 encoded file in place.
+///
+/// Decodes the base64 encoded content of a file in place and writes the decoded content back to the
+/// file. If the file does not contain base64 encoded content, maintains the file as is.
+///
+/// # Arguments
+///
+/// * `filename` - A string slice that holds the path to the file to decode.
+pub fn try_decode_base64_file_inplace(filename: &str) -> Result<()> {
```

Comment:
I have updated the base64 file decoding function for better error handling, could you please check for any incorrect implementations or issues that I may have introduced here?